Property stub based type definition does not support parameter default value

Some of operators allows default value if it's not specified like below,

```
function repeat<T>(count: number = -1);
function defaultIfEmpty<T, R>(defaultValue: R = null);
```

and `Observable` has property stub to create prototype in module like 
`repeat: <T>(count: number) => Observable<T>`
default value for parameter is not allowed, end result causes typescript compiler complains without parameter.

```
Rx.Observable.of(42).defaultIfEmpty();
error TS2346: Supplied parameters do not match any signature of call target.
```

related with https://github.com/ReactiveX/RxJS/issues/608, and https://github.com/ReactiveX/RxJS/pull/643 for remove property based stubs.

--- a/src/CoreOperators.ts
+++ b/src/CoreOperators.ts
@@ -22,7 +22,7 @@ export interface CoreOperators<T> {
   dematerialize?: () => Observable<any>;
   debounce?: (durationSelector: (value: T) => Observable<any> | Promise<any>) => Observable<T>;
   debounceTime?: <R>(dueTime: number, scheduler?: Scheduler) => Observable<R>;
-  defaultIfEmpty?: <R>(defaultValue: R) => Observable<T> | Observable<R>;
+  defaultIfEmpty?: <R>(defaultValue?: T | R) => Observable<T> | Observable<R>;
   delay?: (delay: number, scheduler?: Scheduler) => Observable<T>;
   distinctUntilChanged?: (compare?: (x: T, y: T) => boolean, thisArg?: any) => Observable<T>;
   do?: (next?: (x: T) => void, error?: (e: any) => void, complete?: () => void) => Observable<T>;
@@ -50,8 +50,8 @@ export interface CoreOperators<T> {
   publishBehavior?: (value: any) => ConnectableObservable<T>;
   publishReplay?: (bufferSize: number, windowTime: number, scheduler?: Scheduler) => ConnectableObservable<T>;
   reduce?: <R>(project: (acc: R, x: T) => R, seed?: R) => Observable<R>;
-  repeat?: (count: number) => Observable<T>;
-  retry?: (count: number) => Observable<T>;
+  repeat?: (count?: number) => Observable<T>;
+  retry?: (count?: number) => Observable<T>;
   retryWhen?: (notifier: (errors: Observable<any>) => Observable<any>) => Observable<T>;
   sample?: (notifier: Observable<any>) => Observable<T>;
   sampleTime?: (delay: number, scheduler?: Scheduler) => Observable<T>;

--- a/src/Observable.ts
+++ b/src/Observable.ts
@@ -170,7 +170,7 @@ export default class Observable<T> implements CoreOperators<T>  {
   dematerialize: () => Observable<any>;
   debounce: (durationSelector: (value: T) => Observable<any> | Promise<any>) => Observable<T>;
   debounceTime: <R>(dueTime: number, scheduler?: Scheduler) => Observable<R>;
-  defaultIfEmpty: <R>(defaultValue: R) => Observable<T> | Observable<R>;
+  defaultIfEmpty: <R>(defaultValue?: T | R) => Observable<T> | Observable<R>;
   delay: (delay: number, scheduler?: Scheduler) => Observable<T>;
   distinctUntilChanged: (compare?: (x: T, y: T) => boolean, thisArg?: any) => Observable<T>;
   do: (next?: (x: T) => void, error?: (e: any) => void, complete?: () => void) => Observable<T>;
@@ -198,8 +198,8 @@ export default class Observable<T> implements CoreOperators<T>  {
   publishBehavior: (value: any) => ConnectableObservable<T>;
   publishReplay: (bufferSize: number, windowTime: number, scheduler?: Scheduler) => ConnectableObservable<T>;
   reduce: <R>(project: (acc: R, x: T) => R, seed?: R) => Observable<R>;
-  repeat: (count: number) => Observable<T>;
-  retry: (count: number) => Observable<T>;
+  repeat: (count?: number) => Observable<T>;
+  retry: (count?: number) => Observable<T>;
   retryWhen: (notifier: (errors: Observable<any>) => Observable<any>) => Observable<T>;
   sample: (notifier: Observable<any>) => Observable<T>;
   sampleTime: (delay: number, scheduler?: Scheduler) => Observable<T>;

